jobspec2/types.config.go

```diff
@@ -315,18 +314,32 @@ func (c *jobConfig) EvalContext() *hcl.EvalContext {
 			end := t.SourceRange().End.Byte
 
 			v := string(body[start:end])
-			if i := bytes.IndexByte(body[end:], '}'); i != -1 {
-				v += string(body[end : end+i+1])
-			} else {
-				// fallback for unexpected cases
-				v += "}"
+
+			// find the start of inclusing "${..}" if it's inclused in one; otherwise, wrap it in one
+			quoted := false
+			for i := start - 1; i >= 1; i-- {
+				if body[i] == '{' && body[i-1] == '$' {
+					quoted = true
+					v = string(body[i-1:start]) + v
+					break
+				} else if body[i] != ' ' {
```

Comment:
Should we being using [`unicode.IsSpace`](https://golang.org/pkg/unicode/#IsSpace) to catch other whitespace chars?

Reply:
Testing emprically, only a plain space works. Trying with any other (e.g. `\t`, `\n`, etc) I get an error like the following:

```
input.hcl:2,23-24: Invalid character; This character is not used within the language.
input.hcl:2,23-24: Extra characters after interpolation expression; Expected a closing brace to end the interpolation expression, but found extra characters.
```